packages/botonic-nlp/src/tasks/ner/botonic-ner.ts

```diff
@@ -0,0 +1,180 @@
+import { ModelEvaluation } from '../../model/types'
+import { Codifier } from '../../preprocess/codifier'
+import { PADDING_TOKEN, UNKNOWN_TOKEN } from '../../preprocess/constants'
+import { Preprocessor } from '../../preprocess/preprocessor'
+import { trainTestSplit } from '../../preprocess/selection'
+import {
+  Normalizer,
+  PaddingPosition,
+  Stemmer,
+  Stopwords,
+  Tokenizer,
+} from '../../preprocess/types'
+import { VocabularyGenerator } from '../../preprocess/vocabulary-generator'
+import { ModelStorage } from '../../storage/model-storage'
+import { Locale } from '../../types'
+import { createBiLstmModel } from '../ner/models/bilstm-model'
+import { NerModelParameters, NerModelTemplate } from './models/types'
+import { NEUTRAL_ENTITY } from './process/constants'
+import { PredictionProcessor } from './process/prediction-processor'
+import { Processor } from './process/processor'
+import { Entity } from './process/types'
+import { NerConfigStorage } from './storage/config-storage'
+
+export class BotonicNer {
+  entities: string[]
+  vocabulary: string[]
+  private processor: Processor
+  private predictionProcessor: PredictionProcessor
+  private modelManager: ModelManager
+
+  private constructor(
+    readonly locale: Locale,
+    readonly maxLength: number,
+    private preprocessor: Preprocessor
+  ) {}
+
+  static with(locale: Locale, maxLength: number): BotonicNer {
+    return new BotonicNer(
+      locale,
+      maxLength,
+      new Preprocessor(locale, maxLength)
+    )
+  }
+
+  static async load(path: string): Promise<BotonicNer> {
+    const config = NerConfigStorage.load(path)
+    const ner = new BotonicNer(
+      config.locale,
+      config.maxLength,
+      new Preprocessor(config.locale, config.maxLength)
+    )
+    ner.vocabulary = config.vocabulary
+    ner.entities = config.entities
+    ner.modelManager = new ModelManager(await ModelStorage.load(path))
+    return ner
+  }
+
+  loadDataset(path: string): Dataset {
+    const dataset = DatasetLoader.load(path)
+    this.entities = Array.from(
+      new Set([NEUTRAL_ENTITY].concat(dataset.entities))
+    )
+    return dataset
+  }
+
+  splitDataset(
+    dataset: Dataset,
+    testSize = 0.2,
+    shuffle = true
+  ): { train: Sample[]; test: Sample[] } {
+    return trainTestSplit(dataset.samples, testSize, shuffle)
+  }
+
+  generateVocabulary(samples: Sample[]): void {
+    this.vocabulary = Array.from(
+      new Set(
+        [PADDING_TOKEN, UNKNOWN_TOKEN].concat(
+          VocabularyGenerator.using(this.preprocessor).generate(samples)
+        )
+      )
+    )
+  }
+
+  compile(): void {
+    this.processor = new Processor(
+      this.preprocessor,
+      new Codifier(this.vocabulary, { categorical: false }),
+      new Codifier(this.entities, { categorical: true })
+    )
+    this.predictionProcessor = new PredictionProcessor(this.entities)
+  }
+
+  async createModel(
+    template: NerModelTemplate,
```

Comment:
i'm a supporter of having these params referenced, so when calling them it's more easy to understand.
instead of:
```
await ner.createModel('biLstm', await DatabaseManager.with('en', 'glove', 50))
```
use it like:
```
await ner.createModel({template: 'biLstm', embeddingsManager:  await DatabaseManager.with('en', 'glove', 50)})
````